### DISCUSSION TITLE
Java app with FluidSynth for Win+Linux+Mac ?

### DISCUSSION THREAD

**Discussion by jjazzboss** [FluidSynth/fluidsynth#1087] [Q&A] [answered]
My java app will generate Midi data and embed the FluidSynth lib to render the music on Win10 x64. I use the FluidSynth 2.2.6 DLLs via the new Java 17 Foreign Function & Memory API and it works great so far.

I don't know well Linux and MacOS so could you confirm that, in order to port the app, I normally just have to replace the Win-x64 DLLs by the corresponding native libs (I guess linux-x64, mac-x64, Mac-aarch64 ?)

Can I assume that by default FluidSynth will send its output on the computer's default audio device ? (so it will be transparent for the end user)

I'm also a bit lost how to get the native libs on Linux... It seems I have to find a x64 Linux computer with Ubuntu and "sudo apt-get install fluidsynth" then retrieve the .so files ? 

Thanks for your help.

**Comment by derselbst** [comment, marked as answer]
> I don't know well Linux and MacOS so could you confirm that, in order to port the app, I normally just have to replace the Win-x64 DLLs by the corresponding native libs

I don't know for sure, but this sounds reasonable.

> Can I assume that by default FluidSynth will send its output on the computer's default audio device ? (so it will be transparent for the end user)

It depends whether or not you're using one of fluidsynth's audio drivers. And which fluidsettings you've provided for those audio drivers.

> I'm also a bit lost how to get the native libs on Linux... It seems I have to find a x64 Linux computer with Ubuntu and "sudo apt-get install fluidsynth" then retrieve the .so files ?

You could probably use the precompiled binaries from any distro. You just need to make sure to also include the .so files fluidsynth depends on. As well as the dependencies of those dependency-libs. Use `ldd libfluidsynth.so`  to check that.

**Reply by jjazzboss** [reply]
Ok thanks. 

Regarding the audio driver, I just create one using `new_fluid_audio_driver(settings, synth)` with default audio settings, like in this [example code](https://www.fluidsynth.org/api/example_8c-example.html). I assumed that FluidSynth would pick the most appropriate one...
